### PR TITLE
Remove unnecessary dependence for profiler

### PR DESCRIPTION
**Pull request by chengduoZH**

### CODE REVIEW

File: paddle/fluid/platform/profiler.h [hunk 1]
```diff
@@ -17,55 +17,12 @@ limitations under the License. */
 #include <list>
 #include <string>
 #include <vector>
-#include "paddle/fluid/platform/device_context.h"
-
+#include "paddle/fluid/platform/enforce.h"
+#include "paddle/fluid/platform/event.h"
+#include "paddle/fluid/platform/gpu_info.h"
```
Comment (luotao1) [inline paddle/fluid/platform/profiler.h:23]:
```
#ifdef PADDLE_WITH_CUDA
#include "paddle/fluid/platform/gpu_info.h"
#endif
```

**Review by luotao1 — COMMENTED** [review]
I test in `-DWITH_GPU=OFF`:
```
[ 19%] Built target test_paddle_inference_api
/home/luotao/Paddle/paddle/fluid/platform/profiler_test.cc: In member function ‘virtual void RecordEvent_RecordEvent_Test::TestBody()’:
/home/luotao/Paddle/paddle/fluid/platform/profiler_test.cc:36:27: error: ‘paddle::platform::DeviceContext’ has not been declared
   using paddle::platform::DeviceContext;
                           ^
make[2]: *** [paddle/fluid/platform/CMakeFiles/profiler_test.dir/build.make:63: paddle/fluid/platform/CMakeFiles/profiler_test.dir/profiler_test.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:4114: paddle/fluid/platform/CMakeFiles/profiler_test.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

File: paddle/fluid/platform/profiler.h (hunk 1 above)
Comment (chengduoZH) [reply paddle/fluid/platform/profiler.h:23]: Done

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**